src/m365/spo/commands/hubsite/hubsite-get.ts

```diff
@@ -25,42 +26,105 @@ class SpoHubSiteGetCommand extends SpoCommand {
     return 'Gets information about the specified hub site';
   }
 
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.url = typeof args.options.url !== 'undefined';
+    return telemetryProps;
+  }
+
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
     spo
       .getSpoUrl(logger, this.debug)
       .then((spoUrl: string): Promise<any> => {
-        const requestOptions: any = {
-          url: `${spoUrl}/_api/hubsites/getbyid('${encodeURIComponent(args.options.id)}')`,
-          headers: {
-            accept: 'application/json;odata=nometadata'
-          },
-          responseType: 'json'
-        };
-
-        return request.get(requestOptions);
+        if (args.options.id) {
+          return this.getHubSiteById(spoUrl, args.options);
+        }
+        else {
+          return this.getHubSite(spoUrl, args.options);
+        }
       })
       .then((res: any): void => {
         logger.log(res);
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
+  private getHubSiteById(spoUrl: string, options: Options): Promise<HubSite> {
+    const requestOptions: any = {
+      url: `${spoUrl}/_api/hubsites/getbyid('${options.id}')`,
+      headers: {
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+    
+    return request
+      .get(requestOptions)
+      .then((hubsite: any) => hubsite)
+      .catch(() => Promise.reject(`The specified hub site with id ${options.id} does not exist`));
```

Comment:
You're making an assumption here that every possible error indicates that the hub does not exist. Let's just return the error from the API so that it bubbles up to the `handleRejectedODataJsonPromise` function.